ios/ios/src/Web Services/CMAPICredentials.m

```diff
@@ -31,6 +32,8 @@ - (void)setAppIdentifier:(NSString *)appId apiKey:(NSString *)apiKey andBaseURL:
     self.appIdentifier = appId;
     self.apiKey = apiKey;
     self.baseURL = baseURL;
+
+    [CMLegacyCacheCleaner cleanLegacyCache];
```

Comment:
Seems like a strange place to kick this process off. Can we use something more top-level in the app lifecycle, such as the app delegate?